opencog/query/PatternUtils.cc

```diff
@@ -117,21 +116,18 @@ void get_connected_components(
 			size_t nc = components.size();
 			for (size_t i = 0; i<nc; i++)
 			{
-				std::set<Handle> cur_vars(component_vars[i]);
```

Comment:
It seems to me (without thinking too deep), what we would need is 2 types of Handles,
```
Handle
```
for handle pointing to atoms that may be modified
```
ConstHandle
```
for atoms that may not be modified. A bit like iterator and const_iterator in STL.